### PR TITLE
Pull Request for major changes to Tests and then the small change in tvl2018 that fixes the issue in signal segment to spectrum

### PR DESCRIPTION
**Pull request by jerhsiao**

### CODE REVIEW

**Comment by MalcolmSlaney** [comment]
Ok, that is a lot of changes.

I thought it was mostly working and you were going to add one more test to show the stuff I broke.  Perhaps just compute the values in the text summary.

It looks like you did that part differently than I did.  Did you find my error?

What do the new tests do?  (I’ll have to port them to the JAX version after we finalize this PR.)

Can you tell me more what is changing?

Thank you!

- Malcolm

**Comment by jerhsiao** [comment]
It is working!
The main program code is mostly the same except for a few lines of code. I described what changed in my last email. I think the error had to do with using numpy ones and zeros, which were messing up the scale somehow. It might have been selecting the wrong window. I changed it to booleans that are set to a specific window and it is more closely aligned to the original MATLAB code.

As for the tests, they look like a lot of changes, but I kept interpolation tests and other small function tests the same, and just moved things around. Previously, I think the passing threshold was too high; it was just checking shapes and low level functionality. I wanted to add tests that could actually be useful for future users if they want to make more changes to the code.

Basically what I did was I found the expected loudness values for the set input of 50 max dbl, 32k rate, 1khz100ms with free field transform. I also found other expected values for each bigger function like signal segment to spectrum or the other excitation functions.
Then I added tests to check the current value against this expected. It looks like a lot but it’s pretty basic and is useful if a user makes changes and wants to compare to a version they know is correct.

Let me know if this makes sense.
Thanks!
Sincerely,
Jeremy Hsiao


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
